chiron/multistate.py

```diff
@@ -0,0 +1,701 @@
+
+    @property
+    def n_replicas(self) -> int:
+        """
+        Get the number of replicas in the sampler.
+
+        Returns
+        -------
+        int
+            The number of replicas.
+        """
+        if self._sampler_states is None:
+            return 0
+        else:
+            return len(self._sampler_states)
+
+    @property
+    def iteration(self):
+        """The integer current iteration of the simulation (read-only).
+
+        If the simulation has not been created yet, this is None.
+
+        """
+        return self._iteration
+
+    @property
+    def mcmc_sampler(self):
+        """A copy of the MCMCSampler used to propagate the simulation.
+
+        This can be set only before creation.
+
+        """
+        import copy
+
+        return copy.deepcopy(self._mcmc_sampler)
+
+    @property
+    def sampler_states(self) -> Optional[List[SamplerState]]:
+        """
+        Get a copy of the sampler states list at the current iteration.
+
+        This property can only be set before running the simulation.
+
+        Returns
+        -------
+        Optional[List[SamplerState]]
+            The list of sampler states at the current iteration, or None if not set.
+        """
+        if self._sampler_states is None:
+            return None
+        import copy
+
+        return copy.deepcopy(self._sampler_states)
+
+    @property
+    def is_periodic(self):
+        """
+        Determine if the system is periodic.
+
+        Returns
+        -------
+        Optional[bool]
+            True if the system is periodic, False if not, and None if not initialized.
+        """
+        if self._sampler_states is None:
+            return None
+        return self._thermodynamic_states[0].is_periodic
+
+    @property
+    def is_completed(self):
+        """Check if we have reached any of the stop target criteria (read-only)"""
+        return self._is_completed()
+
+    def _compute_replica_energies(self, replica_id: int) -> np.ndarray:
+        """
+        Compute the energy of a replica across all thermodynamic states.
+
+        Parameters
+        ----------
+        replica_id : int
+            The index of the replica for which to compute energies.
+
+        Returns
+        -------
+        np.ndarray
+            An array of energies for the replica across all thermodynamic states.
+        """
+        from chiron.states import calculate_reduced_potential_at_states
+
+        # Retrieve sampler state associated to this replica.
+        sampler_state = self._sampler_states[replica_id]
+        # Compute energy for all thermodynamic states.
+        energies = calculate_reduced_potential_at_states(
+            sampler_state, self._thermodynamic_states, self.nbr_list
+        )
+        return energies
+
+    def create(
+        self,
+        thermodynamic_states: List[ThermodynamicState],
+        sampler_states: List[SamplerState],
+        nbr_list: NeighborListNsqrd,
```

Comment:
```suggestion
        nbr_list: Optional[List[PairsBase]]=None,
```

Should also have this be the PairsBase for the typing instead of NeighborListNsqrd.